File: src/painel.cpp
#include "painel.hpp"
#include <iomanip>
#include <sstream>
#include <format>

std::string linhaPainel(const Sensor& sensor) {
    // TODO ETAPA 01: consultar somente o contrato Sensor.
    std::ostringstream streamValor;
    streamValor << std::fixed << std::setprecision(1) << sensor.valor();

    return sensor.tag() + ": " + streamValor.str() + " " + sensor.unidade() + " | " + (sensor.emAlerta() ? "ALERTA" : "OK");
}

File: src/painel.py
from sensores import Sensor


def linha_painel(sensor: Sensor) -> str:
    # TODO ETAPA 01: consultar somente o contrato Sensor.
    return f"{sensor.tag}: {sensor.valor():.1f} {sensor.unidade()} | {'ALERTA' if sensor.em_alerta() else 'OK'}"
